### PR TITLE
fix(ngcc): handle presence of both `ctorParameters` and `__decorate`

### PR DESCRIPTION
**Pull request by JoostK**
Recently ng-packagr was updated to include a transform that used to be
done in tsickle (https://github.com/ng-packagr/ng-packagr/pull/1401),
where only constructor parameter decorators are emitted in tsickle's
format, not any of the other decorators.

ngcc used to extract decorators from only a single format, so once it 
saw the `ctorParameters` static property it assumed the library is using
the tsickle format. Therefore, none of the `__decorate` calls were
considered. This resulted in missing decorator information, preventing 
proper processing of a package.

This commit changes how decorators are extracted by always looking at 
both the static properties and the `__decorate` calls, merging these 
sources appropriately.

Resolves FW-1573

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Is there a typo in the commit message?

> ngcc used to only a single format,

**Review by petebacondarwin — APPROVED** [review]
LGTM - I wonder if there is a risk that tools like ng-packagr may mix such decorator styles even more loosely? In which case, should we just change the code to look for both types of decorator always and then merge the two sets that are discovered? This would make use resilient to such tooling changes in the future. The only downside would be a small performance loss, right?

**Review by gkalpak — APPROVED** [review]
+1 for always looking for both types of decorators and then merging the two sets (unless the performance decrease is significant).

Otherwise lgtm2 :+1:

**Comment by JoostK** [comment]
I thought about merging the two and decided not to, as it would require some merging strategy that may depend yet again on how tooling generates code. However I see now that before the refactoring in #31614 we were essentially already merging the two formats, so it makes sense to start doing that again.

**Comment by JoostK** [comment]
@petebacondarwin I fixed the commit message and pushed a fixup commit so that the two decorator sources are now merged, preferring static properties.

Regarding performance, I don't expect it to matter that much. Recall that for non-tsickle libraries we were already executing both code paths.

File: packages/compiler-cli/ngcc/src/host/esm2015_host.ts [hunk 1]
```diff
@@ -655,12 +655,16 @@ export class Esm2015ReflectionHost extends TypeScriptReflectionHost implements N
       return this.decoratorCache.get(decl) !;
     }
 
-    // First attempt extracting decorators from static properties.
-    let decoratorInfo = this.computeDecoratorInfoFromStaticProperties(classSymbol);
-    if (decoratorInfo === null) {
-      // If none were present, use the `__decorate` helper calls instead.
-      decoratorInfo = this.computeDecoratorInfoFromHelperCalls(classSymbol);
-    }
+    // Extract decorators from static properties and `__decorate` helper calls, then merging them
```
Comment (petebacondarwin) [inline packages/compiler-cli/ngcc/src/host/esm2015_host.ts]: Typo:

```suggestion
    // Extract decorators from static properties and `__decorate` helper calls, then merge them
```

**Review by petebacondarwin — APPROVED** [review]

**Comment by naveedahmed1** [comment]
I hope this will be added in next release  :)

**Comment by JoostK** [comment]
@naveedahmed1 It should be available in `9.0.0-next.9` right about now! :D

**Comment by naveedahmed1** [comment]
Awesome, thank you @JoostK for the update :)

**Pull request closed without merging**